pico_w/wifi/dtls/certs/makecerts.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/bash
+set -e
+
+if [ "${PWD##*/}" != "certs" ]; then
+    echo Run this in the certs folder
+    exit 1
+fi
+if [ -z "$DTLS_SERVER" ]; then
+    echo Define DTLS_SERVER
+    exit 1
+fi
+if ! command -v openssl 2>&1 >/dev/null; then
+    echo openssl could not be found
+    exit 1
+fi
+
+SERVER_NAME=$DTLS_SERVER
+
+if [ -d "$SERVER_NAME" ]; then
+    echo Run \"rm -fr $SERVER_NAME\" to regenerate these keys
+    exit 1
+fi
+mkdir $SERVER_NAME
+echo Generating keys in $PWD/$SERVER_NAME
+
+openssl genrsa -out $SERVER_NAME/ca.key 2048
+openssl req -new -x509 -days 99999 -key $SERVER_NAME/ca.key -out $SERVER_NAME/ca.crt -subj "/C=UK/ST=Cambridgeshire/L=Cambridge/O=Raspberry Pi Ltd/OU=Software/CN=rpiroot"
+
+openssl genrsa -out $SERVER_NAME/server.key 2048
+openssl req -new -out $SERVER_NAME/server.csr -key $SERVER_NAME/server.key -subj "/C=UK/ST=Cambridgeshire/L=Cambridge/O=Raspberry Pi Ltd/OU=Software/CN=$SERVER_NAME"
+openssl x509 -req -in $SERVER_NAME/server.csr -CA $SERVER_NAME/ca.crt -CAkey $SERVER_NAME/ca.key -CAcreateserial -out $SERVER_NAME/server.crt -days 9999
+
+openssl genrsa -out $SERVER_NAME/client.key 2048
+openssl req -new -out $SERVER_NAME/client.csr -key $SERVER_NAME/client.key -subj "/C=UK/ST=Cambridgeshire/L=Cambridge/O=Raspberry Pi Ltd/OU=Software/CN=$SERVER_NAME"
+openssl x509 -req -in $SERVER_NAME/client.csr -CA $SERVER_NAME/ca.crt -CAkey $SERVER_NAME/ca.key -CAcreateserial -out $SERVER_NAME/client.crt -days 999
+
+echo -n \#define DTLS_ROOT_CERT \" > $SERVER_NAME/dtls_client.inc
+cat $SERVER_NAME/ca.crt | awk '{printf "%s\\n\\\n", $0}' >> $SERVER_NAME/dtls_client.inc
+echo -e "\"\n" >> $SERVER_NAME/dtls_client.inc
+
+echo -n \#define DTLS_KEY \" >> $SERVER_NAME/dtls_client.inc
+cat $SERVER_NAME/client.key | awk '{printf "%s\\n\\\n", $0}' >> $SERVER_NAME/dtls_client.inc
+echo -e "\"\n" >> $SERVER_NAME/dtls_client.inc
+
+echo -n \#define DTLS_CERT \" >> $SERVER_NAME/dtls_client.inc
+cat $SERVER_NAME/client.crt | awk '{printf "%s\\n\\\n", $0}' >> $SERVER_NAME/dtls_client.inc
+echo -e "\"\n" >> $SERVER_NAME/dtls_client.inc
+
+echo -n \#define DTLS_ROOT_CERT \" > $SERVER_NAME/dtls_server.inc
+cat $SERVER_NAME/ca.crt | awk '{printf "%s\\n\\\n", $0}' >> $SERVER_NAME/dtls_server.inc
+echo -e "\"\n" >> $SERVER_NAME/dtls_server.inc
+
+echo -n \#define DTLS_KEY \" >> $SERVER_NAME/dtls_server.inc
+cat $SERVER_NAME/server.key | awk '{printf "%s\\n\\\n", $0}' >> $SERVER_NAME/dtls_server.inc
+echo -e "\"\n" >> $SERVER_NAME/dtls_server.inc
+
+echo -n \#define DTLS_CERT \" >> $SERVER_NAME/dtls_server.inc
+cat $SERVER_NAME/server.crt | awk '{printf "%s\\n\\\n", $0}' >> $SERVER_NAME/dtls_server.inc
+echo "\"" >> $SERVER_NAME/dtls_server.inc
```

Comment:
This makes it a bit easier to see what `makecerts.sh` is actually doing :slightly_smiling_face: 
```suggestion
function append_file_define {
  DEFINE=$1
  INFILE=$2
  OUTFILE=$3
  echo -n \#define $DEFINE \" >> $OUTFILE
  awk '{printf "%s\\n\\\n", $0}' $INFILE >> $OUTFILE
  echo -e "\"\n" >> $OUTFILE
}

append_file_define DTLS_ROOT_CERT $SERVER_NAME/ca.crt $SERVER_NAME/dtls_client.inc
append_file_define DTLS_KEY $SERVER_NAME/client.key $SERVER_NAME/dtls_client.inc
append_file_define DTLS_CERT $SERVER_NAME/client.crt $SERVER_NAME/dtls_client.inc

append_file_define DTLS_ROOT_CERT $SERVER_NAME/ca.crt $SERVER_NAME/dtls_server.inc
append_file_define DTLS_KEY $SERVER_NAME/server.key $SERVER_NAME/dtls_server.inc
append_file_define DTLS_CERT $SERVER_NAME/server.crt $SERVER_NAME/dtls_server.inc
```